makefile

```diff
@@ -30,12 +30,20 @@ include make/libraries
 ##
 # Set default compiler options.
 ##
-CFLAGS = -I . -isystem $(EIGEN) -isystem $(BOOST) -isystem $(VIENNACL)  -isystem$(CVODES)/include -Wall -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DNO_FPRINTF_OUTPUT -pipe -DVIENNACL_WITH_OPENCL -fPIC -lOpenCL
+CFLAGS = -I . -isystem $(EIGEN) -isystem $(BOOST) -isystem $(VIENNACL)  -isystem$(CVODES)/include -Wall -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DNO_FPRINTF_OUTPUT -pipe -fPIC 
 CFLAGS_GTEST = -DGTEST_USE_OWN_TR1_TUPLE 
 LDLIBS =
 EXE =
 WINE =
 
+##
+# Check for STAN_GPU flag
+##
+ifneq (,$(findstring STAN_GPU, $(CFLAGS)))
```

Comment:
Daniel had suggested we use a makefile variable instead of just -D STAN_GPU, which would make this stuff a little simpler. I'm checking with Ben if that makes sense for RStan and will get back to you.